src/box/lua/experimental/connpool.lua

```diff
@@ -92,7 +92,70 @@ local function get_candidates(role_name, opts)
     return candidates
 end
 
+local function get_connection(role_name, opts)
+    checks('string', {
+        labels = '?table',
+        prefer_local = '?boolean',
+    })
+    opts = opts or {}
+
+    local candidates_opts = {
+        labels = opts.labels
+    }
+    local candidates = get_candidates(role_name, candidates_opts)
+    if next(candidates) == nil then
+        if opts.labels ~= nil then
+            local msg = 'No instances with role %q and labels %s are available'
+            error(msg:format(role_name, table.concat(opts.labels)), 0)
+        end
+        local msg = 'No instances with role %q are available'
+        error(msg:format(role_name), 0)
+    end
+
+    if opts.prefer_local == nil or opts.prefer_local == true then
+        local name = box.info.name
+        for n, candidate_name in pairs(candidates) do
+            if candidate_name == name then
+                table.remove(candidates, n)
```

Comment:
We shoudn't modify a table during `pairs()`.

Reply:
BTW, this code would be simpler if we just check `candidates` using a helper like `table_has_key` (that I proposed in comments for one of the previous patches).

Reply:
True, fixed. For now I did not implemented `table_has_key()`.